coderd/workspaceagents_test.go

```diff
@@ -445,13 +445,19 @@ func TestWorkspaceAgentTailnet(t *testing.T) {
 	_ = agenttest.New(t, client.URL, authToken)
 	resources := coderdtest.AwaitWorkspaceAgents(t, client, ws.ID)
 
-	ctx, cancelFunc := context.WithCancel(context.Background())
-	defer cancelFunc()
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+	defer cancel()
 	conn, err := client.DialWorkspaceAgent(ctx, resources[0].Agents[0].ID, &codersdk.DialWorkspaceAgentOptions{
 		Logger: slogtest.Make(t, nil).Named("client").Leveled(slog.LevelDebug),
 	})
 	require.NoError(t, err)
 	defer conn.Close()
+
+	// Connection should remain open even if the dial context is canceled.
+	cancel()
```

Comment:
nit: I'm wondering if you can extract this to a separate function to make sure that `cancel()` is called only once in defer, instead of adding an extra `cancel()` call.

Reply:
It doesn't really matter if it's called multiple times (it's no-op for context cancel funcs). I typically always `defer` as a precaution to ensure it's cleaned up, then call it manually when it's time accepting that there will be a double cancel (this is especially important when using `require` that can short-circuit a test).

Reply:
> It doesn't really matter if it's called multiple times 

I know, I'm looking at this `cancel()` juggling more from the clean code perspective.

Reply:
Alright, refactored 👍🏻